runtime/src/iree-amd-aie/driver/xrt/pipeline_layout.h

```diff
@@ -0,0 +1,64 @@
+// Copyright 2023 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#ifndef IREE_AMD_AIE_DRIVER_XRT_PIPELINE_LAYOUT_H_
+#define IREE_AMD_AIE_DRIVER_XRT_PIPELINE_LAYOUT_H_
+
+#include "iree/base/api.h"
+#include "iree/hal/api.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif  // __cplusplus
+
+//===----------------------------------------------------------------------===//
+// iree_hal_xrt_descriptor_set_layout_t
+//===----------------------------------------------------------------------===//
+
+// Creates a descriptor set layout for the given |bindings|.
+//
+// |out_descriptor_set_layout| must be released by the caller (see
+// iree_hal_descriptor_set_layout_release).
+iree_status_t iree_hal_xrt_descriptor_set_layout_create(
+    iree_allocator_t host_allocator,
```

Comment:
Similar here: when we have `host_allocator`, the convention is to place it right before what it allocates.